File: helpers/components/serverMessage.js
const serverMessage = (key, data = []) => {
  let error = false;
  let message;
  let status = 500;

  switch (key) {
    case "PHONE":
      error = true;
      status = 400;
      message = "MISSING_PHONE_NUMBER";
      break;

    case "FIRST_NAME":
      error = true;
      status = 400;
      message = "MISSING_FIRST_NAME";
      break;

    case "LAST_NAME":
      error = true;
      status = 400;
      message = "MISSING_LAST_NAME";
      break;

    case "NO_PASSWORD":
      error = true;
      status = 400;
      message = "MISSING_PASSWORD";
      break;

    case "EMPTY_FIELD":
      error = true;
      status = 400;
      message = "FIELD_ARE_REQUIRED";
      break;

    case "WRONG_PASSWORD":
      error = true;
      status = 406;
      message = "WRONG_PASSWORD";
      break;

    case "USER_EXIST":
      error = true;
      status = 401;
      message = "USER_ALREADY_EXIST";
      break;

    case "PHONE_EXIST":
      error = true;
      status = 401;
      message = "PHONE_NUMBER_ALREADY_TAKED";
      break;

    case "ADMIN_EXIST":
      error = true;
      status = 401;
      message = "ADMIN_ALREADY_EXIST";
      break;

    case "REGISTRATION_FAILD":
      error = true;
      status = 401;
      message = "USER_REGISTRATION_FAILD";
      break;

    case "AUTH_FAILED":
      error = true;
      status = 417;
      message = "USER_NOT_FOUND";
      break;

    case "REGISTRED":
      error = false;
      status = 200;
      message = "SUCCESSFUL_REGISTRATION";
      break;

    case "LOGIN":
      error = false;
      status = 200;
      message = "SUCCESSFULLY_AUTHENTICATED";
      break;

    case "FAILED_DELETTION":
      error = true;
      status = 417;
      message = "DELETTION_REQUEST_FAILED";
      break;

    case "PROMOTION_ERROR":
      error = true;
      status = 401;
      message = "ERROR_IN_USER_PROMOTION";
      break;

    case "PROMOTED":
      error = false;
      status = 401;
      message = "USER_HAS_BEEN_PROMOTED";
      break;

    case "DELETED":
      error = false;
      status = 200;
      message = "SUCCESSFULLY_DELETED";
      break;

    case "REVOCATION_FAILED":
      error = true;
      status = 401;
      message = "REVOCATION_OPPERATION_FAILED";
      break;

    case "REVOKED":
      error = false;
      status = 401;
      message = "USER_HAS_BEEN_REVOKED";
      break;

    case "AUTHORIZED":
    case "SUCCESS":
      error = false;
      status = 200;
      message = "SUCCESS";
      break;

    case "REQUEST_LOAN":
      error = true;
      status = 401;
      message = "LOAN_NOT_AUTHORIZED_AT_THE_MOMENT";
      break;

    case "NOT_AUTHORIZED":
      error = true;
      status = 401;
      message = "NOT_AUTHORIZED_REQUEST";
      break;

    case "ACCESS_DENIED":
      error = true;
      status = 401;
      message = "ACCESS_DENIED";
      break;

    case "ERROR_LOAN":
      error = true;
      status = 401;
      message = "LOAN_ADDING_FAILD";
      break;

    case "LOAN_ADDED":
      error = false;
      status = 201;
      message = "LOAN_ADDED_SUCCESSFULLY";
      break;

    case "ERROR_SERVER":
    default:
      error = true;
      status = status;
      message = "INTERNAL_SERVER_ERROR";
      break;
  }

  return { error, status, message, data };
};

module.exports = { serverMessage };

File: lib/helpers/components/server_message.dart
import 'dart:developer';

import 'package:confetti/confetti.dart';
import 'package:get/get.dart';

import '../index.dart';

ConfettiController conffetiController =
    ConfettiController(duration: const Duration(seconds: 10));
serverMessage(
  String key,
) {
  String message;
  String title;

  switch (key) {
    case "MISSING_PHONE_NUMBER":
      message =
          "Phone number is a required field for continuation. Please provide this information to proceed.";
      showAlertDialog(text: message);
      break;

    case "MISSING_FIRST_NAME":
      message =
          "First name is a required field for continuation. Please provide this information to proceed.";
      showAlertDialog(text: message);
      break;

    case "MISSING_LAST_NAME":
      message =
          "Last name is a required field for continuation. Please provide this information to proceed.";
      showAlertDialog(text: message);
      break;

    case "MISSING_PASSWORD":
      message =
          "Password is a required field for continuation. Please provide this information to proceed.";
      showAlertDialog(text: message);
      break;

    case "WRONG_PASSWORD":
      message =
          "Incorrect Password. Please enter the correct password to proceed.";
      showAlertDialog(text: message);
      break;

    case "USER_ALREADY_EXIST":
      message =
          "User Already Exists. Please choose a different data or log in with your existing credentials.";
      showAlertDialog(text: message);
      break;

    case "PHONE_NUMBER_ALREADY_TAKED":
      message =
          "User Already Exists. Please choose a different phone number or log in with your existing credentials.";
      showAlertDialog(text: message);
      break;

    case "ADMIN_ALREADY_EXIST":
      message =
          "Admin Already Exists. Please choose a different data or log in with your existing credentials.";
      showAlertDialog(text: message);
      break;

    case "USER_REGISTRATION_FAILD":
      title = "Registration Failed.";
      message =
          "Unable to complete user registration. Please try again or contact support for assistance.";
      showAlertDialog(text: title, subTitle: message);
      break;

    case "USER_NOT_FOUND":
      title = "User Not Found.";
      message = "Please check your credentials and try again.";
      showAlertDialog(text: title, subTitle: message);
      break;

    case "SUCCESSFUL_REGISTRATION":
      conffetiController.play();
      title = "Successful Registration";
      message = "An email confirmation has been sent to you for verification.";
      Future.delayed(const Duration(seconds: 8), () {
        conffetiController.stop();
      });
      showAlertDialog(
          assetName: "validate.json",
          repeat: false,
          size: 80,
          text: title,
          subTitle: message);
      break;

    case "SUCCESSFULLY_AUTHENTICATED":
      message = "Successfully Authenticated. Welcome back!";
      showAlertDialog(
        assetName: "validate.json",
        isOnlyValidator: true,
      );

      Future.delayed(
        const Duration(seconds: 5),
        () {
          Get.back();
          log("Login page");
        },
      );
      break;

    case "DELETTION_REQUEST_FAILED":
      message = "Deletion Request Failed. Please try again.";
      showAlertDialog(text: message);
      break;

    case "ERROR_IN_USER_PROMOTION":
      title = "Error in User Promotion.";
      message = "Please contact support for assistance.";
      showAlertDialog(text: title, subTitle: message);
      break;

    case "USER_HAS_BEEN_PROMOTED":
      message = "User has been promoted successfully.";
      showAlertDialog(
        assetName: "validate.json",
        repeat: false,
        size: 80,
        text: message,
      );
      break;

    case "SUCCESSFULLY_DELETED":
      message = "Successfully deleted.";
      showAlertDialog(
        assetName: "validate.json",
        repeat: false,
        size: 80,
        text: message,
      );
      break;

    case "REVOCATION_OPPERATION_FAILED":
      message = "Revocation Operation Failed. Please try again.";
      showAlertDialog(text: message);
      break;

    case "USER_HAS_BEEN_REVOKED":
      message = "User has been revoked successfully.";
      showAlertDialog(
        assetName: "validate.json",
        repeat: false,
        size: 80,
        text: message,
      );
      break;

    case "AUTHORIZED":
    case "SUCCESS":
      message = "Operation Successful.";
      showAlertDialog(
        assetName: "validate.json",
        repeat: false,
        size: 80,
        text: message,
      );
      break;

    case "NOT_AUTHORIZED_REQUEST":
      message = "Not Authorized Request. Access denied.";
      showAlertDialog(text: message);
      break;

    case "ACCESS_DENIED":
      message = "Access Denied. You do not have permission for this request.";
      showAlertDialog(text: message);
      break;

    case "ERROR_SERVER":
    case "INTERNAL_SERVER_ERROR":
    default:
      title = "Internal Server Error.";
      message = "Please try again later or contact support for assistance.";
      showAlertDialog(text: title, subTitle: message);
      break;
  }
}
